services/server/src/handlers/lesson.ts

```diff
@@ -75,11 +79,28 @@ function create(context: ApiContext) {
       const slot = await availabilitySlots.findById(payload.slotId);
       if (!slot) return next(notfound.slot());
 
+      // the booked lesson should be in the future
+      if (dayjs.utc(payload.start).isBefore(dayjs.utc())) {
+        return next(bad());
+      }
+
       const price = calculateLessonPrice(
         platformConfig.tutorHourlyRate,
         payload.duration
       );
 
+      // check if the user already has a lesson currently or in future
+      // with this tutor and return conflict if so
+      const curLessons = await lessons.find({
+        users: [user.id],
+        after: dayjs.utc().toISOString(),
+        canceled: false,
+      });
+      if (curLessons.total !== 0) {
+        return next(conflict());
```

Comment:
- Let's return a specific error for this case and call it `reachedBookingLimit`. This will be used by the frontned to show the end user a meaningful error message. 
- Let's also rename `conflict` error to:
    - `conflictingLessons` -> use it when there you cannot book the lesson. 
    - `conflictingSchedule` -> use it with tutor incase his schedule is malformed. 

```suggestion
        return next(conflict());
```